tests/system_tests/test_blueapi_system.py

```diff
@@ -0,0 +1,214 @@
+
+    get_task = client.get_task(created_task.task_id)
+    assert (
+        get_task.task_id == created_task.task_id
+        and get_task.is_pending
+        and not get_task.is_complete
+        and len(get_task.errors) == 0
+    )
+
+    client.clear_task(created_task.task_id)
+
+
+def test_get_non_existent_task(client: BlueapiClient):
+    with pytest.raises(KeyError) as exception:
+        client.get_task("Not-exists")
+        assert str(exception) == "{'detail': 'Item not found'}"
+
+
+def test_delete_non_existent_task(client: BlueapiClient):
+    with pytest.raises(KeyError) as exception:
+        client.clear_task("Not-exists")
+        assert str(exception) == "{'detail': 'Item not found'}"
+
+
+def test_put_worker_task(client: BlueapiClient):
+    created_task = client.create_task(_SIMPLE_TASK)
+    client.start_task(WorkerTask(task_id=created_task.task_id))
+    active_task = client.get_active_task()
+    assert active_task.task_id == created_task.task_id
+    client.clear_task(created_task.task_id)
+
+
+def test_put_worker_task_fails_if_not_idle(client: BlueapiClient):
+    small_task = client.create_task(_SIMPLE_TASK)
+    long_task = client.create_task(_LONG_TASK)
+
+    client.start_task(WorkerTask(task_id=long_task.task_id))
+    active_task = client.get_active_task()
+    assert active_task.task_id == long_task.task_id
+
+    with pytest.raises(BlueskyRemoteControlError) as exception:
+        client.start_task(WorkerTask(task_id=small_task.task_id))
+        assert str(exception) == "<Response [409]>"
+    client.abort()
+    client.clear_task(small_task.task_id)
+    client.clear_task(long_task.task_id)
+
+
+def test_get_worker_state(client: BlueapiClient):
+    assert client.get_state() == WorkerState.IDLE
+
+
+def test_set_state_transition_error(client: BlueapiClient):
+    with pytest.raises(BlueskyRemoteControlError) as exception:
+        client.resume()
+        assert str(exception) == "<Response [400]>"
+
+    with pytest.raises(BlueskyRemoteControlError) as exception:
+        client.pause()
+        assert str(exception) == "<Response [400]>"
+
+
+def test_get_task_by_status(client: BlueapiClient):
+    task_1 = client.create_task(_SIMPLE_TASK)
+    task_2 = client.create_task(_SIMPLE_TASK)
+    task_by_pending_request = requests.get(
+        client._rest._url("/tasks"), params={"task_status": TaskStatusEnum.PENDING}
+    )
+    assert task_by_pending_request.status_code == status.HTTP_200_OK
+    task_by_pending = TypeAdapter(TasksListResponse).validate_python(
+        task_by_pending_request.json()
+    )
+
+    assert len(task_by_pending.tasks) == 2
+    for task in task_by_pending.tasks:
+        trackable_task = TypeAdapter(TrackableTask).validate_python(task)
+        assert trackable_task.is_complete is False and trackable_task.is_pending is True
+
+    client.start_task(WorkerTask(task_id=task_1.task_id))
+    while not client.get_task(task_1.task_id).is_complete:
+        ...
```

Comment:
Maybe put a `time.sleep(0.1)` here, just so it doesn't check 1000s of times per second, same for the one below.